app/models/alchemy/node.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class Node < BaseRecord
+    VALID_URL_REGEX = /\A(\/|\D[a-z\+\d\.\-]+:)/
```

Comment:
What does the `:` at the end do here?

Reply:
Ahh. It's the `http:` or `email:` end-of-the-protocol part of the URL. Yeah, makes sense, the validation can't be too strict, otherwise we'd get false negatives which would annoy users a lot. :D

Reply:
> Ahh. It's the http: or email: end-of-the-protocol part of the URL.

Exactly. 

> the validation can't be too strict, otherwise we'd get false negatives

it shouldn't be. do you have an example of an annoying false negative? www domains instead of http: ?

Reply:
I think we're in agreement here. I'm saying we can't try validating the entire URL, as the stuff after the protocol can literally by anything: `my-crazy-scheme:takes?🦊🦊emojigreat!*&` I get the reasoning.

Reply:
Oh, yeah. Absolutely. 

Remind me to add a proper error message to that validation 👍